### PR TITLE
Add NYSL license, rule and test files #400

### PR DESCRIPTION
**Pull request by techytushar**
PR for issue #400 
Signed-off-by: Tushar Mittal <chiragmittal.mittal@gmail.com>

### CODE REVIEW

**Comment by pombredanne** [comment]
Ah got it, you opened a new PR ;) You could have reused the previous one, FWIW.

File: src/licensedcode/data/rules/nysl-0.9982.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+licenses:
+    - nysl-0.9982
```
Comment (pombredanne) [inline src/licensedcode/data/rules/nysl-0.9982.yml]: Please add a trailing linefeed

File: tests/licensedcode/data/licenses/nysl-0.9982.yml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+licenses:
+    - nysl-0.9982
```
Comment (pombredanne) [inline tests/licensedcode/data/licenses/nysl-0.9982.yml]: Please add a trailing linefeed

File: tests/licensedcode/data/licenses/nysl-0.9982.txt [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+B. The author is not responsible for any kind of damages or loss
+  while using or misusing this software, which is distributed
+  "AS IS". No warranty of any kind is expressed or implied.
+  You use AT YOUR OWN RISK.
+
+C. Copyrighted to <copyright holders>
+
+D. Above three clauses are applied both to source and binary
+  form of this software.
```
Comment (pombredanne) [inline tests/licensedcode/data/licenses/nysl-0.9982.txt]: Looking good! could you also add two extra tests? One with only the english test and one with only the japanese text?

Comment (techytushar) [reply tests/licensedcode/data/licenses/nysl-0.9982.txt]: sure 👍

Comment (pombredanne) [reply tests/licensedcode/data/licenses/nysl-0.9982.txt]: use sensible file names: the name does not matter, this is just a convention so:
- nysl-0.9982_en.txt and nysl-0.9982_en.yml  
- nysl-0.9982_jp.txt and nysl-0.9982_jp.yml  

would work well

File: src/licensedcode/data/licenses/nysl-0.9982.yml [hunk 4]
```diff
@@ -0,0 +1,10 @@
+key: nysl-0.9982
+short_name: NYSL 0.9982
+name: Software-License-NYSL-v0.9982
+category: Copyleft Limited
```
Comment (techytushar) [inline src/licensedcode/data/licenses/nysl-0.9982.yml]: @DennisClark  hey, can you please check the license category?

Comment (pombredanne) [reply src/licensedcode/data/licenses/nysl-0.9982.yml]: @DennisClark any opinion?

**Comment by pombredanne** [comment]
Thank you for this.
The tests are failing though:
did you try to run
`py.test -vvs tests/licensedcode -k test_detection_nysl` locally to see what fails in details?

**Comment by techytushar** [comment]
i guess when i run the tests the nysl-0.9982_en and the nysl-0.9982_jp files gives error... 

screenshot from 2018-02-10 23-57-59

screenshot from 2018-02-10 23-57-52

File: src/licensedcode/data/rules/nysl-0.9982.RULE [hunk 5]
```diff
@@ -0,0 +1 @@
+http://www.kmonos.net/nysl/index.ja.html
```
Comment (yash-nisar) [inline src/licensedcode/data/rules/nysl-0.9982.RULE]: Why do we have a URL here ?

Comment (yash-nisar) [reply src/licensedcode/data/rules/nysl-0.9982.RULE]: IIRC, we explicitly have the text in the `.RULE` file. You can have a look at the other files for reference. :)

File: tests/licensedcode/data/licenses/nysl-0.9982_en.yml [hunk 6]
```diff
@@ -0,0 +1,2 @@
+licenses:
+    - nysl-0.9982_en
```
Comment (yash-nisar) [inline tests/licensedcode/data/licenses/nysl-0.9982_en.yml]: This line could be the cause of the `AssertionError`. Try replacing this with `nysl-0.9982`

File: tests/licensedcode/data/licenses/nysl-0.9982_jp.yml [hunk 7]
```diff
@@ -0,0 +1,2 @@
+licenses:
+    - nysl-0.9982_jp
```
Comment (yash-nisar) [inline tests/licensedcode/data/licenses/nysl-0.9982_jp.yml]: Maybe replace this with `nysl-0.9982` too.

File: src/licensedcode/data/rules/nysl-0.9982.RULE (hunk 5 above)
Comment (techytushar) [reply src/licensedcode/data/rules/nysl-0.9982.RULE]: i looked at the other `.RULE` files they all had the URL of the license homepage.. :)

Comment (yash-nisar) [reply src/licensedcode/data/rules/nysl-0.9982.RULE]: Can you point me to a few of them ?

Comment (techytushar) [reply src/licensedcode/data/rules/nysl-0.9982.RULE]: `bigdigits_1.RULE`, `biopython.RULE`, `attribution.RULE`, `artistic-2.0_2.RULE`. Some of them also contain the license text, but most of them contains the URL.

Comment (yash-nisar) [reply src/licensedcode/data/rules/nysl-0.9982.RULE]: Thanks @techytushar  👍

Comment (techytushar) [reply src/licensedcode/data/rules/nysl-0.9982.RULE]: :+1:

**Comment by pombredanne** [comment]
Thanks! I merged this with https://github.com/nexB/scancode-toolkit/commit/b1d6d6d11eed7cbc676643f7b9424f1c07de6584

**Pull request closed without merging**